src/api/posts.js

```diff
@@ -0,0 +1,37 @@
+import { BASE_URL } from './api';
+
+export const getUserPosts = () => fetch(`${BASE_URL}/posts`)
+  .then(response => response.json());
```

Comment:
you don't filtering posts by user there, so it's rather getAllPosts, or getPosts, not getUserPosts